controllers/machinehealthcheck_controller.go

```diff
@@ -145,7 +145,7 @@ func (r *MachineHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.R
 	defer func() {
 		// Always attempt to patch the object and status after each reconciliation.
 		// Patch ObservedGeneration only if the reconciliation completed successfully
-		patchOpts := []patch.Option{}
+		var patchOpts []patch.Option
```

Comment:
Why is this preferred? These two lines are functionally equivalent, I think this is very much a style preference. Which is most used throughout the codebase?

Reply:
I think that's the most common argumentation: https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices

Reply:
According to go lint the second one is preferred because there is no memory allocation (just declaring a slice that is nil)